File: SDKs/github.com/api.github.com/1.1.4/python/src/sdk/models/operations/packages_get_package_for_organization.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from sdk.models import shared


@dataclass
class PackagesGetPackageForOrganizationPathParams:
    org: str = field(metadata={'path_param': { 'field_name': 'org', 'style': 'simple', 'explode': False }})
    package_name: str = field(metadata={'path_param': { 'field_name': 'package_name', 'style': 'simple', 'explode': False }})
    package_type: shared.PackageTypeEnum = field(metadata={'path_param': { 'field_name': 'package_type', 'style': 'simple', 'explode': False }})
    

@dataclass
class PackagesGetPackageForOrganizationRequest:
    path_params: PackagesGetPackageForOrganizationPathParams = field()
    

@dataclass
class PackagesGetPackageForOrganizationResponse:
    content_type: str = field()
    status_code: int = field()
    package: Optional[shared.Package] = field(default=None)
    

File: SDKs/github.com/api.github.com/1.1.4/typescript/src/sdk/models/operations/packagesgetpackagefororganization.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PackagesGetPackageForOrganizationPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=org" })
  org: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=package_name" })
  packageName: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=package_type" })
  packageType: shared.PackageTypeEnum;
}


export class PackagesGetPackageForOrganizationRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PackagesGetPackageForOrganizationPathParams;
}


export class PackagesGetPackageForOrganizationResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  package?: shared.Package;
}
